pkg/controller/elasticsearch/driver/downscale.go

```diff
@@ -213,7 +213,7 @@ func doDownscale(downscaleCtx downscaleContext, downscale ssetDownscale, actualS
 	}
 
 	// Expect the updated statefulset in the cache for next reconciliation.
-	downscaleCtx.expectations.ExpectGeneration(downscale.statefulSet.ObjectMeta)
+	downscaleCtx.expectations.ExpectGeneration(downscale.statefulSet)
```

Comment:
Why we need to do it directly? Could we just call `ReconcileStatefulSet` instead of `Update` above? This would call ExpectGeneration for us and be consistent with upscale code path.

Reply:
I think you're right 👍 
If you don't mind I'll create a follow-up issue for that: https://github.com/elastic/cloud-on-k8s/issues/2028
I think in the process we may want to move the template hash computation thing in the `ReconcileStatefulSet` function.